src/evict/evict_page.c

```diff
@@ -151,6 +151,9 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint8_t previous_state, uint32
         __wt_evict_list_clear_page(session, ref);
     }
 
+    /* We have exclusive access, invalidate any weak hazard pointers. */
+    __wt_hazard_weak_invalidate(session, ref);
```

Comment:
Do we want to do this as close to the call to `__wt_reconcile` as possible? There are a number of places where `__evict_review` will return instead of calling reconciliation and now we've invalidated any number of still valid weak references.

Reply:
I added a comment saying we will do the work of identifying the best place to invalidate later in the project.